feature/experimental/system/gnmi/benchmarking/ate_tests/drained_configuration_convergence_time/drained_configuration_convergence_time_bgp_test.go

```diff
@@ -246,10 +251,26 @@ func verifyBGPSetMED(t *testing.T, dut *ondatra.DUTDevice, ate *ondatra.ATEDevic
 			// Validate if all prefixes are received by ATE.
 			isConverged(t, dut, ate, ap)
 			rib := at.NetworkInstance(ap.Name()).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "0").Bgp().Rib()
-			prefixPath := rib.AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Ipv4Unicast().
-				NeighborAny().AdjRibInPre().RouteAny().WithPathId(0).Prefix()
-			pref := gnmi.GetAll(t, ate, prefixPath.State())
-			gotSetMED := gnmi.GetAll(t, ate, rib.AttrSetAny().Med().State())
+			routeP := rib.AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Ipv4Unicast().
+				NeighborAny().AdjRibInPre().RouteAny().WithPathId(0)
+			routes := gnmi.GetAll(t, ate, routeP.State())
+			attrs := gnmi.GetAll(t, ate, rib.AttrSetAny().State())
+			mask := net.IPv4Mask(255, 255, 255, 0)
+			masked := net.ParseIP(setup.AdvertiseBGPRoutesv4).Mask(mask)
+			var gotSetMED []uint32
+			var pref []string
+			for _, route := range routes {
+				ip, _, _ := net.ParseCIDR(route.GetPrefix())
+				pref = append(pref, route.GetPrefix())
+				if ip.Mask(mask).Equal(masked) {
+					idx := route.GetAttrIndex()
+					if idx >= uint64(len(attrs)) {
+						t.Errorf("Invalid attr-index %d for prefix: %s", idx, route.GetPrefix())
+						continue
+					}
+					gotSetMED = append(gotSetMED, attrs[idx].GetMed())
```

Comment:
Please add length of gotSetMED should be equal to routeCount.  ?

Reply:
This is already being done in cmp.Diff. The `wantSetMED` list is of length equal to routeCount. 

So if `gotSetMED` is not of the same length, the test would throw error.